### ISSUE TITLE
DBeaver data export runs without errors but is incomplete

### ISSUE THREAD

**Issue by camillahandley** [dbeaver/dbeaver#21883]
### Description

Before I export the data from my SQL query results, I can see the total row count is ~2 million. I export to CSV with all the default settings (i.e., fetch size = 10000) and the resulting file does not include all the data. I've tried multiple times and each time I get a different row count, so I cannot detect a pattern, but it is only a small percentage of the data (e.g. 50k rows). There are no errors that appear on the Error Log. My coworker was able to run the same query and export the full dataset on his computer. He has a Mac and was running a older version of DBeaver.

### DBeaver Version

Community Edition Version 23.2.4

### Operating System

Windows 10 Enterprise Version 22H2 64bit OS

### Database and driver

Dremio JDBC driver 
DREMIO-Dev database

### Steps to reproduce

1. I run a query from the DREMIO-Dev database that results in a large data table (2 million rows)
2. I click 'Calculate Total Row Count' to confirm the size of the data table
3. I click "Export Data" and the Data Transfer window pops up 
4. I select 'Export to CSV file(s)', then Next >
5. I leave the Extraction Settings at the default, then click Next >
6. I leave Format settings at the default, then click Next >
7. I designate the folder path and file name on the Output page, then click Next >
8. I click Proceed and wait for the query to run again. The query runs in much less time than expected and the resulting CSV file does not include all the rows. 

I have attached a picture of the Source settings and Target settings 

![image](https://github.com/dbeaver/dbeaver/assets/59592274/43a4954f-fff6-4f96-b497-7e3ff12f29fa)

**Comment by LonwoLonwo** [comment]
Hello @camillahandley 

Could you please provide the log files? You can find instructions here: https://dbeaver.com/docs/wiki/Log-files/. Thank you.

**Comment by camillahandley** [comment]
Here are the log files. I tried to export the data multiple times on 11/16/2023. I tried a couple times this morning and it worked and gave me the full dataset but I didn't change anything. I am connected to my work's network via a VPN today. 
[.log](https://github.com/dbeaver/dbeaver/files/13394599/default.log)

**Comment by hrimthrusa** [comment]
Hi @camillahandley 
Thank you for such a thorough steps-to-reproduce section. 

My main concern is the step 8: you don't have to query anything after you pressed **_Proceed_** button, you just have to wait for the loading to complete. During the loading, the resulting file is incomplete if opened.

I couldn't reproduce it with a 64M rows Dremio database using DBeaver bundled connector in the latest DBeaver version 23.2.5. Please try to reproduce it after you upgrade. Here is the link: https://dbeaver.io/download/ or follow Help (main toolbar) -> Check for updates.

Also, please try to rerun the process with **Open new connections** and **Select row count parameters** checked (Data Export -> Extraction settings step) in the latest version.

**Comment by camillahandley** [comment]
Hi @hrimthrusa 

Thanks for your response. I updated to DBeaver 23.2.5 and checked those two parameters in the Extraction settings step and got the same results. My query had 1.5M rows, but the exported CSV had 22k rows. I tried a second time, doing all the same things and it exported the full dataset. 

To clarify, in step 8 I am not doing anything after pressing the **Proceed** button, what I mean to say is that the query that runs after you press the **Proceed** button to export the data goes much faster than expected. I don't open the file until the export task is complete.

**Comment by hrimthrusa** [comment]
@camillahandley Hey there.

Here are some other ideas worth trying.

Make sure you **don't** have the _**Split output file**_ parameter checked in the _**Output**_ step during the export tuning.

Remove the **_${scriptFileName}_** tab from your naming pattern as it is deprecated in DBeaver. 
Change the behaviour during the _data_ or _value file name conflict_ event to **_Ask_**.
Also, try to change the directory to a new folder on your drive C to make sure you are not writing into an administrated folder with a limited access. Or even to another drive (D?) should you have any. (chances are thin but never zero).

Basically, put your export settings in accordance with mine:
![gitHub](https://github.com/dbeaver/dbeaver/assets/97725836/db0d8456-839b-4abf-b4a8-c249202a243d)

**Comment by camillahandley** [comment]
@hrimthrusa Thank you for these tips. I was able to match your settings except for the behavior for blob value file name conflict (for some reason I cannot change that setting) and the exports have been consistently working. 

Not sure where the problem was exactly, but I think we found a solution. Thank you!

**Comment by arhayka** [comment]
Sounds great! 
We cannot reproduce the issue with your settings, so I'm closing it.
